From 27f0bd5a3339612e03112e6b490900a9fabc3337 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 9 Aug 2022 11:44:25 -0700
Subject: [PATCH 5/5] stackdriver: Fix return type mismatch

Fix
error: incompatible integer to pointer conversion returning 'int' from a function with result type 'flb_sds_t' (aka 'char *') [-Wint-conversion]
            return -1;
                   ^~

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 plugins/out_stackdriver/stackdriver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/out_stackdriver/stackdriver.c b/plugins/out_stackdriver/stackdriver.c
index ae66bf2..e01755c 100644
--- a/plugins/out_stackdriver/stackdriver.c
+++ b/plugins/out_stackdriver/stackdriver.c
@@ -2033,7 +2033,7 @@ static flb_sds_t stackdriver_format(struct flb_stackdriver *ctx,
             flb_sds_destroy(operation_producer);
             msgpack_unpacked_destroy(&result);
             msgpack_sbuffer_destroy(&mp_sbuf);
-            return -1;
+            return NULL; 
         }
 
         /* Number of parsed labels */
-- 
2.37.1

